.github/workflows/ci_tests.yml

```diff
@@ -193,8 +239,9 @@ jobs:
   run_integrated_tests:
     needs:
       - is_not_draft_pull_request
+      - get_integrated_test_labels
       - cpu_builds
-    if: "${{ contains( fromJSON( needs.is_not_draft_pull_request.outputs.LABELS ), 'ci: run integrated tests') || github.event_name != 'pull_request' }}"
+    if: "${{ (contains( fromJSON( needs.get_integrated_test_labels.outputs.LABELS ), 'ci: run integrated tests') || github.event_name != 'pull_request') }}"
```

Comment:
Do you then need to check that the labels indeed contain `ci: run integrated tests`?
If you're here, then `get_integrated_test_labels` did not fail, which means that `ci: run integrated tests` was found?
Would it make the job if you rename `get_integrated_test_labels` into `are_integrated_tests_required` and let `are_integrated_tests_required` fail if not? And then you can remove the `outputs` from your guard jobs.

Same for CUDA.

Reply:
Good point, I think you are totally right. Should be fixed now.